spec/gemstash/http_client_spec.rb

```diff
@@ -20,6 +20,16 @@
     @other_server.stop
   end
 
+  describe "#for" do
+    it "sets configured timeout value" do
+      configured_timeout = 99
+      client = Gemstash::HTTPClient.for(Gemstash::Upstream.new(@server.url), configured_timeout)
+      expect(
+        client.instance_variable_get(:@client).options.timeout
+      ).to eq(configured_timeout)
+    end
```

Comment:
Could this be tested as a behaviour rather than as cracking the object open by reflection to check that the internal have a specific shape?

What I mean is that in this file, at the beginning we are setting up a SimpleServer, we could have a "slow mount" that takes something like 0.2 seconds sleep and then setup the upstream with a timeout of 0.1, that should be enough to make it timeout. Then a second test with a timeout of 0.3 should be enough.

Consider that I am making times up and that it could be a bit harder to fine tune the testing, but I would be much more happier checking behaviour than implementation.